core/src/transfer.rs

```diff
@@ -246,6 +254,8 @@ impl Transaction {
     }
 
     /// Return the contract call data, if there is any.
+    ///
+    /// Returns `None` if there is no contract call to a non-genesis contract.
```

Comment:
I would change it to remove the genesis ref
Indeed, transfer::convert is a call to a genesis contract, and this function will return Some